### DISCUSSION TITLE
Design a good way to access levelflip without scripting

### DISCUSSION THREAD

**Discussion by Narre** [SuperTux/supertux#2074] [Ideas]
When #2042 is merged, flipping will (hopefully) be mostly polished and ready for use. With that in mind, I think it would be great to have a somewhat intuitive way to flip the level without using scripts, since not every level creator is comfortable with or even capable of (doesn't understand) scripting. Besides, it would be great to have a ***consistent*** way to flip levels that players could recognise and know what's about to happen (to avoid situations where one level designer uses buttons, one powerups, one random scripttriggers etc).

**Edit:** I have just realised the potion already flips the level. I've no idea when this was added or if it was always that way (if so, then I apologise for being stupid). Either way, I'm leaving this open since a bit of discussion can't hurt. Please feel free to laugh at me for not knowing and/or checking this before.

**Comment by Narre** [comment]
**Suggestion:** Make either `levelflip()` or `Level.flip_vertically()` (which one is up for discussion) the default behaviour of the red potion (**edit:** the latter already is, maybe discuss if it should be the former instead, or if there should be a unique graphical transition effect for flipping), and reserve the blue potion for a potential future mirroring (horizontal flipping) feature, as suggested in #863. This wouldn't affect existing levels using the potions for other uses since a powerup's script always has priority over its default behaviour.

**Comment by Narre** [comment]
**Suggestion:** Add a new sprite for a powerup specifically designed for level flipping, just like the suggestion above, except with a new sprite. **Edit:** pointless now that I realised the potion already does that, I'll leave this comment here just to discuss how the sprite should look.

**Reply by Rusty-Box** [reply]
I had an idea for updating the potion sprite by adding something like circling arrows to show what it does.

**Comment by Narre** [comment]
**Suggestion:** Add a completely new object specifically designed to trigger levelflip.

**Reply by Narre** [reply]
**Downside:** This would require some extra coding and I'm not sure whether a completely new object for this purpose isn't an unnecessary overkill.

**Reply by Narre** [reply]
**Upside:** This would allow for the implementation of some creative ideas when designing said object and could potentially lead to more creative ways of using the flipping feature.
